server/src/main/java/org/apache/druid/catalog/model/ColumnSpec.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.catalog.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import org.apache.druid.guice.annotations.UnstableApi;
+import org.apache.druid.java.util.common.IAE;
+
+import javax.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Specification of table columns. Columns have multiple types
+ * represented via the type field.
+ */
+@UnstableApi
+public class ColumnSpec
+{
+  private final String type;
+  private final String name;
+  private final String sqlType;
```

Comment:
I agree that `ColumnType` is the actual Druid type. The `ColumnType` is the storage type, which is limited. For example, all complex types are represented as `UKNOWN_COMPLEX`, which is the physical, not user view. Over time, we'll want an actual type name for a complex type: something like `IP_ADDRESS` rather than just "complex". The specific names are an active point of discussion. Also, a `__time` is a `LONG` internally, but it is a `TIMESTAMP` semantically.

We do limit the set of SQL types: only those that directly map to Druid types.

Reply:
I think the descriptions of the fields given here should go directly into javadoc so that the documentation lives forever in the code, and not only in a comment thread.